### PR TITLE
Add the ability to delete a template

### LINKED ISSUE #12742: Add ability to delete a template
Unblocked by https://github.com/influxdata/influxdb/issues/12824

may need to regenerate client
add delete wrapper to templates in client
add delete button to templates list which deletes template by id

### PR DESCRIPTION
**Pull request by Palakp41**
Co-authored-by: Alirie Gray <alirie.gray@gmail.com>

Closes #12742 

_Briefly describe your proposed changes:_
Updated the client to match the swagger's definition of delete template
Updated template card to call Delete action
Created actions and reducers to update the redux state to remove the template

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/templates/components/TemplateImportOverlay.tsx [hunk 1]
```diff
@@ -58,16 +52,12 @@ class TemplateImportOverlay extends PureComponent<Props> {
...
-      await createTemplate(template)
-      notify(importTemplateSucceeded())
-      setTemplatesStatus(RemoteDataState.NotStarted)
-    } catch (error) {
-      notify(importTemplateFailed(error))
-    }
+
+    const template = JSON.parse(importString)
+    await createTemplate(template)
+    setTemplatesStatus(RemoteDataState.NotStarted)
```
Comment (ebb-tide) [inline ui/src/templates/components/TemplateImportOverlay.tsx]: awesome, much better thanks!

File: ui/src/templates/reducers/index.ts [hunk 2]
```diff
@@ -59,6 +59,17 @@ const templatesReducer = (
...
       }
+
+      case ActionTypes.RemoveTemplateSummary: {
+        const {templateID} = action.payload
+        const {items} = draftState
+        const deleted = items.filter(l => {
+          return l.id !== templateID
+        })
+
+        draftState.items = deleted
```
Comment (ebb-tide) [inline ui/src/templates/reducers/index.ts]: the name deleted feels a bit misleading, could we set `draftState.items = items.filter .....` directly?

File: ui/src/templates/components/TemplateImportOverlay.tsx (hunk 1 above)
Comment (Palakp41) [reply ui/src/templates/components/TemplateImportOverlay.tsx]: 👍

File: ui/src/templates/reducers/index.ts (hunk 2 above)
Comment (Palakp41) [reply ui/src/templates/reducers/index.ts]: I fixed that :)

**Pull request merged**